lib/util/package-rules/current-version.ts

```diff
@@ -20,7 +20,7 @@ export class CurrentVersionMatcher extends Matcher {
     }
     const isUnconstrainedValue =
       !!lockedVersion && is.nullOrUndefined(currentValue);
-    const version = allVersioning.get(versioning);
+    const version = allVersioning.get(versioning ? versioning : undefined);
```

Comment:
```suggestion
    if (is.nullOrUndefined(versioning)) {
      logger.debug(`No versioning found for package rule matchCurrentVersion '${matchCurrentVersion}'`);
      return null;
    }
    const version = allVersioning.get(versioning);
```

Why not simply returning if versioning is undefined?

Reply:
looks like a config error and we should also issue a warning?

Reply:
If it's undefined then it is defaulted to semver-coerced today. Null does not fall back though.

Returning false here would be wrong

Reply:
returning null would ignore that rule and falsify the result. 

IMO we should rather change this here: 
https://github.com/renovatebot/renovate/blob/c8fceec593a184cc7b858bb1186dd52c6a1206ed/lib/modules/versioning/index.ts#L19

```diff
- export function get(versioning = ''): VersioningApi {
+ export function get(versioning = defaultVersioning.id): VersioningApi { 
```

Reply:
This change wouldn't coerce null, right? See my updated code